pytket_qir/qir.py

```diff
@@ -400,11 +471,76 @@ def condition_zero_block():
                 one=lambda: condition_one_block(),
                 zero=lambda: condition_zero_block(),
             )
+        elif isinstance(op, WASMOp):
+            cregs = _retrieve_registers(circ.bits, BitRegister)
+            args = command.args
+
+            inputs: List[str] = []
+            outputs: List[str] = []
+            for reglist, sizes in [
+                (inputs, op.input_widths),
+                (outputs, op.output_widths),
+            ]:
+                for in_width in sizes:
+                    assert in_width >= 0
+                    wasm_bits = args[:in_width]
+                    args = args[in_width:]
+                    regname = wasm_bits[0].reg_name
```

Comment:
This won't work if `in_width == 0`.

Reply:
Yes, but isn't it exactly what we want ? Or am I getting confused ? We don't want to proceed any further if the WASMOp carries empty registers ?

Reply:
True enough. I am a bit confused about it tbh. What was wrong with `in_width > 0` ? I guess it caught the right case ? Ie. we don't want to process further `WASMOp`s that carry over empty registers ?

Reply:
As far as I know, a `WASMOp` with 0 input registers (but >0 output registers) is valid: it still performs some computation and outputs a (fixed) answer. If this is the case, we should handle it. But I haven't actually checked. If in fact it is invalid, we should throw a descriptive exception.

Reply:
Ah OK, thanks I see. Will do.

Reply:
OK after a quick check, it turns out that the only invalid WASM case is when there are no inputs not outputs, this means that `in_width == 0` should be handled.

Reply:
Done.